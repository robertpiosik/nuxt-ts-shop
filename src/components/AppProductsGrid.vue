<template>
  <transition-group :class="$style.grid" :move-class="$style.move" tag="div">
    <slot />
  </transition-group>
</template>

<script lang="ts">
import { Vue } from 'vue-property-decorator';

export default class ProductsGrid extends Vue {}
</script>

<style lang="scss" module>
.grid {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: auto;
  gap: 1.5rem;
  margin: 1.5rem 0;
  @include ieOnly {
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: repeat(4, auto);
  }
  @include atSmall {
    margin: 3rem 0;
  }
  @include atMedium {
    grid-template-columns: repeat(3, 1fr);
    grid-template-rows: auto;
    @include ieOnly {
      grid-template-columns: repeat(3, 1fr);
      grid-template-rows: repeat(3, auto);
    }
  }
  @include atLarge {
    grid-template-columns: repeat(4, 1fr);
    grid-template-rows: auto;
    @include ieOnly {
      grid-template-columns: repeat(4, 1fr);
      grid-template-rows: repeat(2, auto);
    }
  }
}
.move {
  transition: transform 0.5s ease-in-out;
}
</style>
