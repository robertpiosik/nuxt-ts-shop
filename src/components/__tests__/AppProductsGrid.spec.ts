import { shallowMount } from '@vue/test-utils'
import AppProductsGrid from './../AppProductsGrid.vue'

const slotContent = 'slot content'

const wrapper = shallowMount(AppProductsGrid, {
	slots: { default: slotContent }
})

describe('Component: ProductsGrid', () => {
  describe('GENERAL', () => {
    it('should match a snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should be a Vue instance', () => {
      expect(wrapper.isVueInstance()).toBeTruthy()
    })
  })
  describe('SLOTS', () => {
    it('should render given slot value and nothing else', () => {
      expect(wrapper.text()).toBe(slotContent)
    })
  })
})
