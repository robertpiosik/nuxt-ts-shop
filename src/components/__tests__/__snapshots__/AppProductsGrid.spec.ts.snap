// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ProductsGrid GENERAL should match a snapshot 1`] = `<transition-group-stub tag="div" moveclass="move" class="grid">slot content</transition-group-stub>`;
